src/Storages/MergeTree/KeyCondition.h

```diff
@@ -413,6 +414,9 @@ class KeyCondition
     ColumnIndices key_columns;
     ExpressionActionsPtr key_expr;
     PreparedSets prepared_sets;
+
+    // If true, do not take any function monotonicity into account
```

Comment:
I think it's not only about the monotonicity...

Will it work correctly for expressions like:
```
SELECT count() FROM table
WHERE partition = 123 AND sipHash64(toString(partition)) % 2 = 1
```
?

(When there are multiple conditions. All of them depend only on columns in partition key. But there are some non-analyzable conditions.)

Reply:
There are no tests for this case.

Reply:
Actually we can apply any function transformations to single point range.